bazel/foreign_cc/0002-local-rate-limit-bucket-backport.patch

```diff
@@ -0,0 +1,3422 @@
+-  bool requestAllowed(absl::Span<const RateLimit::LocalDescriptor> request_descriptors);
+-  uint32_t maxTokens(absl::Span<const RateLimit::LocalDescriptor> request_descriptors);
+-  uint32_t remainingTokens(absl::Span<const RateLimit::LocalDescriptor> request_descriptors);
+-  int64_t remainingFillInterval(absl::Span<const RateLimit::LocalDescriptor> request_descriptors);
++  Filters::Common::LocalRateLimit::LocalRateLimiterImpl::Result
++  requestAllowed(absl::Span<const RateLimit::LocalDescriptor> request_descriptors);
+ 
+-  const FilterConfig* getConfig() const;
+   FilterConfigSharedPtr config_;
++  // Actual config used for the current request. Is config_ by default, but can be overridden by
++  // per-route config.
++  const FilterConfig* used_config_{};
++  OptRef<const Filters::Common::LocalRateLimit::TokenBucketContext> token_bucket_context_;
+ 
+-  absl::optional<std::vector<RateLimit::LocalDescriptor>> stored_descriptors_;
+   VhRateLimitOptions vh_rate_limits_;
+ };
+ 
+ } // namespace LocalRateLimitFilter
+ } // namespace HttpFilters
+ } // namespace Extensions
+-} // namespace Envoy
++} // namespace Envoy
+\ No newline at end of file
+diff --git a/source/extensions/filters/listener/local_ratelimit/local_ratelimit.cc b/source/extensions/filters/listener/local_ratelimit/local_ratelimit.cc
+index 7aa4561b80..85f810ad40 100644
+--- source/extensions/filters/listener/local_ratelimit/local_ratelimit.cc
++++ source/extensions/filters/listener/local_ratelimit/local_ratelimit.cc
+@@ -21,7 +21,7 @@ FilterConfig::FilterConfig(
+           Protobuf::RepeatedPtrField<
+               envoy::extensions::common::ratelimit::v3::LocalRateLimitDescriptor>()) {}
+ 
+-bool FilterConfig::canCreateConnection() { return rate_limiter_.requestAllowed({}); }
++bool FilterConfig::canCreateConnection() { return rate_limiter_.requestAllowed({}).allowed; }
+ 
+ LocalRateLimitStats FilterConfig::generateStats(const std::string& prefix, Stats::Scope& scope) {
+   const std::string final_prefix = "listener_local_ratelimit." + prefix;
+diff --git a/source/extensions/filters/network/local_ratelimit/local_ratelimit.cc b/source/extensions/filters/network/local_ratelimit/local_ratelimit.cc
+index 8c6af4c820..09594692dd 100644
+--- source/extensions/filters/network/local_ratelimit/local_ratelimit.cc
++++ source/extensions/filters/network/local_ratelimit/local_ratelimit.cc
+@@ -94,7 +94,7 @@ LocalRateLimitStats Config::generateStats(const std::string& prefix, Stats::Scop
+   return {ALL_LOCAL_RATE_LIMIT_STATS(POOL_COUNTER_PREFIX(scope, final_prefix))};
+ }
+ 
+-bool Config::canCreateConnection() { return rate_limiter_->requestAllowed({}); }
++bool Config::canCreateConnection() { return rate_limiter_->requestAllowed({}).allowed; }
+ 
+ Network::FilterStatus Filter::onNewConnection() {
+   if (!config_->enabled()) {
+diff --git a/test/common/common/BUILD b/test/common/common/BUILD
```

Comment:
the substantial amount of changes in this patch, i would really really like to see the tests compile and passing (as mentioned [here](https://github.com/solo-io/envoy-gloo/pull/357#issuecomment-2253175286)), if that's something that you're able to pull off.

that said, it does look like everything is passing [here](https://github.com/solo-io/gloo/pull/9822), so if we can't get the envoy tests merged and passing here, i think we should be good enough